src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -616,9 +641,9 @@ await foreach (ConfigurationSetting setting in client.GetConfigurationSettingsAs
             return serverData;
         }
 
-        private async Task LoadKeyValuesRegisteredForRefresh(ConfigurationClient client, IDictionary<string, ConfigurationSetting> data, CancellationToken cancellationToken)
+        private async Task<Dictionary<KeyValueIdentifier, ConfigurationSetting>> LoadKeyValuesRegisteredForRefresh(ConfigurationClient client, IDictionary<string, ConfigurationSetting> data, CancellationToken cancellationToken)

```

Comment:
We need to remove the call to LoadKeyValuesRegisteredForRefresh inside LoadSelectedKeyValues